web-api/src/personalization/routes.rs

```diff
@@ -429,8 +436,26 @@ struct DevOption {
 }
 
 impl DevOption {
-    fn is_some(&self) -> bool {
-        self.hybrid.is_some() || self.max_number_candidates.is_some()
+    fn validate(&self, enable_dev: bool, count: usize) -> Result<(), Error> {
+        if !enable_dev && (self.hybrid.is_some() || self.max_number_candidates.is_some()) {
+            // notify the caller instead of silently discarding the dev option
+            return Err(ForbiddenDevOption::DevDisabled.into());
+        }
+        if let Some(DevHybrid::EsRrf { .. }) = self.hybrid {
+            // not available because of the es license, return a 403 instead of forwarding a 500
+            return Err(ForbiddenDevOption::EsRrfUnlisenced.into());
+        }
```

Comment:
:pushpin:  should we then just remove the code?

Reply:
yes

Reply:
i added [ET-4859] to the backlog